### PR TITLE
fix(update): Clarify meaning of --aggressive as --recursive

### PR DESCRIPTION
**Pull request by epage**
When working on cargo-upgrade, I found the meaning of `--aggressive` confusing and named it `--recursive` there.

Renaming this in `cargo update` (with a backwards compatible alias) was referenced in #12425.

### CODE REVIEW

**Comment by epage** [comment]
This was mentioned in #12425 but because it was more of an aside, I want to do a straw poll to make sure people are on board specifically with this before merging as it is a one-way door.

@rfcbot fcp merge

File: src/etc/cargo.bashcomp.sh [hunk 1]
```diff
@@ -87,7 +87,7 @@ _cargo()
 	local opt__t="$opt__test"
 	local opt__tree="$opt_common $opt_pkg_spec $opt_feat $opt_mani $opt_lock --target -i --invert --prefix --no-dedupe --duplicates -d --charset -f --format -e --edges"
 	local opt__uninstall="$opt_common $opt_lock $opt_pkg --bin --root"
-	local opt__update="$opt_common $opt_mani $opt_lock $opt_pkg --aggressive --precise --dry-run"
+	local opt__update="$opt_common $opt_mani $opt_lock $opt_pkg --aggressive --recursive --precise --dry-run"
```
Comment (weihanglo) [inline src/etc/cargo.bashcomp.sh:90]: Should we demote `--aggressive` by removing it from shell completions?

**Review by weihanglo — COMMENTED** [review]
Should we have a simple test to verify that `--aggressive` alias still works?

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks for the update!

@bors r+

**Pull request merged**